### PR TITLE
Configure hive caching

### PR DESCRIPTION
**Pull request by arunthirupathi**
1. Bypass using cached directory result, when caching is disabled.
2. DirectroyListing can choose to disable caching via HiveDirectoryContext.

Test plan 
Existing tests.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/main/java/com/facebook/presto/hive/CachingDirectoryLister.java [hunk 1]
```diff
@@ -78,9 +78,13 @@ public Iterator<HiveFileInfo> list(
             NamenodeStats namenodeStats,
             HiveDirectoryContext hiveDirectoryContext)
     {
-        List<HiveFileInfo> files = cache.getIfPresent(path);
-        if (files != null) {
-            return files.iterator();
+        if (hiveDirectoryContext.isCacheable()) {
```
Comment (abhiseksaikia) [inline presto-hive/src/main/java/com/facebook/presto/hive/CachingDirectoryLister.java:81]: Thanks for fixing this, not having the check was resulting in incorrect result when the session property was set to disable the cache.

**Review by abhiseksaikia — APPROVED** [review]
LGTM!

**Review by NikhilCollooru — APPROVED** [review]
LGTM

**Comment by NikhilCollooru** [comment]
Test failure TestBackgroundHiveSplitLoader.testCachingDirectoryLister() seems to be related to the changes in this PR. Please fix it .

**Pull request merged**